### DISCUSSION TITLE
Best way to format md/mdx with remark

### DISCUSSION THREAD

**Discussion by vrepsys** [remarkjs/remark#1198] [Q&A]
Hey 👋  What's the best way to format md files programatically? Hesitant to use prettier because mdx v2 support isn't there yet.

I'm specifically interested in things like max line length, `remarkStringify` doesn't seem to have a configuration variable for this. I was also looking into `remark-cli` and `remark-lint-maximum-line-length`, but they don't seem to do such formatting, only warnings/linting.

**Comment by ChristianMurphy** [comment]
Heya @vrepsys! 👋 
Recommendations remain the same as https://github.com/orgs/mdx-js/discussions/2067

**Reply by vrepsys** [reply]
I went through that thread, I had an impression the example in the answer is for formatting mdx specific elements only, and doesn't work for paragraphs etc.

 I also wasn't able to format line length of paragraphs using remark-cli, only lint them.

**Reply by ChristianMurphy** [reply]
`remark-lint` itself cannot make fixes, _yet_. https://github.com/remarkjs/remark-lint/issues/82
Some are indirectly fixable through `remark-stringify` options.

**Reply by vrepsys** [reply]
Wondering if anyone found a way to run prettier for markdown elements and leave mdx code untouched, and then format mdx nodes using remark-mdx? Is this feasible?

**Reply by wooorm** [reply]
prettier and remark are formatters. They format. They aren’t supposed to leave stuff untouched.

I also feel like as a JS community we kinda had this discussions that it’s nice to have formatters instead of endless discussions of how to format things manually!
